src/Servers/IIS/src/AspNetCoreModuleV2/CommonLib/BindingInformation.h

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <string>
+#include "ConfigurationSource.h"
+#include "StringHelpers.h"
+#include "WebConfigConfigurationSource.h"
+
+#define CS_SITE_SECTION                         L"system.applicationHost/sites"
+#define CS_SITE_NAME                            L"name"
+#define CS_SITE_BINDINGS                        L"bindings"
+#define CS_SITE_BINDING_INFORMATION             L"bindingInformation"
+#define CS_SITE_BINDING_INFORMATION_ALL_HOSTS   L"*"
+#define CS_SITE_BINDING_PROTOCOL                L"protocol"
+#define CS_SITE_BINDING_PROTOCOL_HTTPS          L"https"
+#define CS_SITE_BINDING_INFORMATION_DELIMITER   L':'
+
+class BindingInformation
+{
+public:
+    BindingInformation(std::wstring protocol, std::wstring host, std::wstring port)
+    {
+        m_protocol = protocol;
+        m_host = host;
+        m_port = port;
+    }
+
+    std::wstring& QueryProtocol()
+    {
+        return m_protocol;
+    }
+
+    std::wstring& QueryPort()
+    {
+        return m_port;
+    }
+
+    std::wstring& QueryHost()
+    {
+        return m_host;
+    }
+
+    static
+    std::vector<BindingInformation>
+    Load(const ConfigurationSource &configurationSource, const IHttpSite& pSite)
+    {
+        std::vector<BindingInformation> items;
+
+        const std::wstring runningSiteName = pSite.GetSiteName();
+
+        auto const siteSection = configurationSource.GetRequiredSection(CS_SITE_SECTION);
+        auto sites = siteSection->GetCollection();
+        for (const auto& site: sites)
+        {
+            auto siteName = site->GetRequiredString(CS_SITE_NAME);
+            if (equals_ignore_case(runningSiteName, siteName))
```

Comment:
Extract parsing logic for IP:PORT:HOST into a separate static method and add unit tests in native.